research/picai/prepare_data.py

```diff
@@ -97,14 +149,39 @@ def preprare_data(
     scans_write_dir: Path,
     annotation_write_dir: Path,
     overview_write_dir: Path,
-    size: Optional[Sequence[int]] = None,
-    physical_size: Optional[Sequence[int]] = None,
-    spacing: Optional[Sequence[int]] = None,
+    size: Optional[Tuple[int, int, int]] = None,
+    physical_size: Optional[Tuple[int, int, int]] = None,
+    spacing: Optional[Tuple[int, int, int]] = None,
     scan_extension: str = "mha",
     annotation_extension: str = ".nii.gz",
     num_threads: int = 4,
     splits_path: Optional[Path] = None,
 ) -> None:
+    """
+    Runs preprocessing on data with specified settings.
+
+    Args:
+        scans_read_dir (Path): The path to read the scans from. Should be a directory with subdirectories
+            for each patient_id. Inside the subdirectories should be all the scan files for a given patient.
+        annotation_read_dir (Path): The path to read the annotations from. Should be a flat directory with all
+            annotation files.
+        scans_write_dir (Path): The path to write the scans to. All scans are written into same directory.
+        annotation_write_dir (Path): The path to write the annotations to.
+            All annotations are written into same directory.
+        overviews_write_dir (Path): The path where the dataset json files are located. For each split 1-5,
+            there is a train and validation file with scan paths, label paths and case labels.
+        size (Optional[Sequence[int]]): Desired dimensions of preprocessed scans in voxels.
+            Triplet of the form: Depth x Height x Width.
+        physical_size (Optional[Sequence[int]]): Desired dimensions of preprocessed scans in mm.
```

Comment:
The types of size, physical_size, spacing can be updated to the tuples type you've changed to.